shell/agents/Microsoft.Azure.Agent/Telemetry.cs

```diff
@@ -0,0 +1,225 @@
+using System.Text.Json;
+
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.WorkerService;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.Agent;
+
+public class AzTrace
+{
+    private static readonly string s_installationId;
+    static AzTrace()
+    {
+        string azureConfigDir = Environment.GetEnvironmentVariable("AZURE_CONFIG_DIR");
+        string userProfile = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+        string userProfilePath = Path.Combine(string.IsNullOrEmpty(azureConfigDir) ? userProfile : azureConfigDir, "azureProfile.json");
```

Comment:
I don't have the env variable `AZURE_CONFIG_DIR` defined, and also don't have either the file `C:\Users\dongbow\azureProfile.json` or the file `C:\Users\dongbow\AzureRmContextSettings.json`. Are you sure we should look at those files and they are the right paths?

Also, we are using AzCLI login token today and the Az PowerShell login token doesn't work. In this case, aren't we supposed to use the AzCLI profile instead?

Reply:
Sorry I make stupid mistakes here. 

Azure CLI and Azure PS share the same installation id. File azureconfig.json contained the id generated by AzCLI Installation. If we are supposed to support Az PowerShell as dependency when in the future the login token is good, we can keep current code.